src/ray/protobuf/job_agent.proto

```diff
@@ -0,0 +1,41 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package ray.rpc;
+
+import "src/ray/protobuf/agent_manager.proto";
+
+message InitializeJobEnvRequest {
+  // The job id.
+  bytes job_id = 1;
```

Comment:
I'm a bit confused about the job_id logic here -- from what I'm gathering here, it looks like the job_id is passed in as part of the job submission spec? Why do we need to do that? Can't we just return the job_id once it's started? That would more closely match the existing functionality.

Reply:
If a `InitializeJobEnvRequest` is received by the job agent, then the network between dashboard and job agent is broken. We don't know the id of the submitted job, we can't query the job status, and we can't drop the job.